google-cloud-spanner/src/test/java/com/google/cloud/spanner/ReadWriteTransactionWithInlineBeginTest.java

```diff
@@ -389,6 +404,71 @@ public void failedBatchUpdateAndThenUpdate() {
     assertThat(countTransactionsStarted()).isEqualTo(2);
   }
 
+  @Test
+  public void executeSqlWithOptimisticConcurrencyControl() {
+    client
+        .readWriteTransaction(Options.optimisticLock())
+        .run(
+            transaction -> {
+              try (ResultSet rs = transaction.executeQuery(SELECT1)) {
+                while (rs.next()) {
+                  assertThat(rs.getLong(0)).isEqualTo(1);
+                }
+              }
+              return null;
+            });
+    Collection<AbstractMessage> requests =
+        Collections2.filter(
+            mockSpanner.getRequests(), msg -> msg.getClass().equals(ExecuteSqlRequest.class));
+    assertThat(requests).hasSize(1);
+    ExecuteSqlRequest request = (ExecuteSqlRequest) Iterables.getOnlyElement(requests);
+    assertThat(request.getTransaction().getBegin()).isEqualTo(OPTIMISTIC_LOCK_OPTIONS);
+  }
+
+  @Test
+  public void readWithOptimisticConcurrencyControl() {
+    client
+        .readWriteTransaction(Options.optimisticLock())
+        .run(
+            transaction -> {
+              try (ResultSet rs =
+                  transaction.read("FOO", KeySet.all(), Collections.singletonList("ID"))) {
+                while (rs.next()) {
+                  assertThat(rs.getLong(0)).isEqualTo(1);
+                }
+              }
+              return null;
+            });
+    Collection<AbstractMessage> requests =
+        Collections2.filter(
+            mockSpanner.getRequests(), msg -> msg.getClass().equals(ReadRequest.class));
+    assertThat(requests).hasSize(1);
+    ReadRequest request = (ReadRequest) Iterables.getOnlyElement(requests);
+    assertThat(request.getTransaction().getBegin()).isEqualTo(OPTIMISTIC_LOCK_OPTIONS);
+  }
+
+  @Test
+  public void beginTransactionWithOptimisticConcurrencyControl() {
+    client
+        .readWriteTransaction(Options.optimisticLock())
+        .run(
+            transaction -> {
+              // This query carries the BeginTransaction, but fails. The BeginTransaction will
+              // then be carried by the subsequent statement.
```

Comment:
This comment is not accurate. Instead of adding the `BeginTransaction` option to the next statement, the client library will force a complete retry of the entire transaction, and use an explicit `BeginTransaction` RPC invocation for that transaction in order to include the failed statement in the transaction as well.

Reply:
Updated the comment, thanks!